basket/src/components/My.tsx

```diff
@@ -0,0 +1,32 @@
+import { useState } from 'react';
+import { Profile } from './Profile';
+import { Login } from './Login';
+
+type Session = {
+  loginUser: {
+    id: number;
+    name: string;
+  };
+  cart: Array<{ id: number; name: string; price: number }>;
+};
+
+type MyProps = {
+  session: Session;
+  login: (id: number, name: string) => void;
+  logout: () => void;
+};
+
+export const My = ({ session, login, logout }: MyProps) => {
+  const [inputValue, setInputValue] = useState('');
+
+  return (
+    <>
+      <div>My</div>
+      {session.loginUser.id !== 0 && session.loginUser.name !== '' ? (
```

Comment:
1. 조건식에 부정문이 들어가면 가독성이 좋지 않기 때문에 가능한 경우라면 긍정문으로 작성하기!
  예시) (부정) 어른이 아니면 복권을 살 수 없어요. vs (긍정) 어린이는 복권을 살 수 없어요.

```suggestion
      {session.loginUser.id && session.loginUser.name ? (
```

2. 조건식이 조금 복잡하다면, 변수로 추출하기


```suggestion
      {isAuthenticated ? (
```

Reply:
1번으로 할 경우, id가 0이거나 name이 ''일 경우도 true로 인식 될 수 있으므로, 2번으로 수정했습니다!